src/cli.pl

```diff
@@ -105,25 +105,23 @@
 getNextVaultAction(VaultAction) :- 
     nl,
     writeln("What would you like to do?"),
-    nl,
     vaultActions(VaultActions),
     getChoice(VaultAction, VaultActions).
 
 
 % Gets user's command choice from the given list
 getChoice(Selected, Choices) :-
-    forall(
-        member(Command,Choices),
-        prettyPrint(Command)
-    ),
-    nl,
-    getInput("Enter number", CommandNumber),
-    % TODO: validate input
-    findall(
-        Atom,
-        member(command(CommandNumber,_,Atom),Choices),
-        [Selected|Rest]
-    ).
+    repeat,
+        nl,
+        forall(
+            member(Command,Choices),
+            prettyPrint(Command)
+        ),
+        nl,
+        getInput("Enter number", CommandNumber),
+    (
+        member(command(CommandNumber,_,Selected),Choices)
```

Comment:
Probably helpful to print an appropriate error message
```suggestion
        member(command(CommandNumber,_,Selected),Choices);
        (writeln("Invalid input - please choose one of the options"), fail)
```